src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs

```diff
@@ -48,14 +48,16 @@ internal struct NameDeclarationInfo
                 var semanticModel = await document.GetSemanticModelForSpanAsync(new Text.TextSpan(token.SpanStart, 0), cancellationToken).ConfigureAwait(false);
                 var typeInferenceService = document.GetLanguageService<ITypeInferenceService>();
 
-                if (IsParameterDeclaration(token, semanticModel, position, cancellationToken, out var result)
+                if (IsTupleElementDeclaration(token, semanticModel, position, cancellationToken, out var result)
```

Comment:
`IsTupleElementDeclaration` needs to come before `IsParameterDeclaration`. Otherwise constructs as 
`void M((System.Array $$` or `(System.Array $$` are treated by `IsParameterDeclaration` and the completion list is empty. This is covered by `TupleElementDefinition6` and `TupleElementDefinition7`.

Reply:
> Otherwise constructs as ```void M((System.Array $$``` [is] treated by IsParameterDeclaration

That itself seems like a bug.  IsParameterDeclaration doesn't seem like it should be true here...

Reply:
> IsParameterDeclaration doesn't seem like it should be true here

This is the syntax tree of `void M((System.Array $$))`

![2018-03-26 10_41_30-roslyn](https://user-images.githubusercontent.com/24472128/37895669-4c49548c-30e2-11e8-8ef0-041d74152b54.png)

So the token is part of a TupleType that itself is a parameter . So @CyrusNajmabadi is right as this is a type declaration taking place in a parameter. This can be solved the way I did (check for tuple declarations before parameter declarations) or by special casing TupleType in IsParameterDeclaration. I'm fine with both ways.